dot/parachain/network-bridge/receiver.go

```diff
@@ -5,33 +5,218 @@ package networkbridge
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"slices"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/ChainSafe/gossamer/dot/network"
+	collatorprotocolmessages "github.com/ChainSafe/gossamer/dot/parachain/collator-protocol/messages"
+	events "github.com/ChainSafe/gossamer/dot/parachain/network-bridge/events"
+	networkbridgemessages "github.com/ChainSafe/gossamer/dot/parachain/network-bridge/messages"
+	validationprotocol "github.com/ChainSafe/gossamer/dot/parachain/validation-protocol"
 
 	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+
+	"github.com/ChainSafe/gossamer/dot/state"
 	"github.com/ChainSafe/gossamer/internal/log"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/crypto/sr25519"
+	"github.com/ChainSafe/gossamer/lib/keystore"
 	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/protocol"
 )
 
 var logger = log.NewFromGlobal(log.AddContext("pkg", "network-bridge"))
 
+var (
+	ErrFinalizedNumber                       = errors.New("finalized number is greater than or equal to the block number")
+	ErrInvalidStringFormat                   = errors.New("invalid string format for fetched collation info")
+	ErrUnexpectedMessageOnCollationProtocol  = errors.New("unexpected message on collation protocol")
+	ErrUnexpectedMessageOnValidationProtocol = errors.New("unexpected message on validation protocol")
+)
+
 type NetworkBridgeReceiver struct {
-	OverseerToSubSystem <-chan any
+	net Network
+
+	BlockState *state.BlockState
+	Keystore   keystore.Keystore
+
+	localView *View
+
+	// Parachains we're currently assigned to. With async backing enabled
+	// this includes assignments from the implicit view.
+	currentAssignments map[parachaintypes.ParaID]uint
+
+	// All active leaves observed by us, including both that do and do not
+	// support prospective parachains. This mapping works as a replacement for
+	// [`polkadot_node_network_protocol::View`] and can be dropped once the transition
+	// to asynchronous backing is done.
+	activeLeaves map[common.Hash]parachaintypes.ProspectiveParachainsMode
+
+	// state tracked per relay parent
+	perRelayParent map[common.Hash]PerRelayParent // map[relay parent]PerRelayParent
+
+	// Collations that we have successfully requested from peers and waiting
+	// on validation.
+	fetchedCandidates map[string]CollationEvent
+	// heads are sorted in descending order by block number
+	liveHeads []parachaintypes.ActivatedLeaf
+
+	finalizedNumber uint32
+
+	OverseerToSubSystem  <-chan any
```

Comment:
Do we really need an `OverseerToSubsystem` channel?
We would get this channel as a argument of Run method